runtime/vm/FFITypeHelpers.hpp

```diff
@@ -248,6 +248,89 @@ class FFITypeHelpers
 		return n;
 	}
 
+	/**
+	 * @brief obtain the ffi_type from the layout symbol (the starting letter of the layout type)
+	 * @param layoutSymb[in] The layout symbol describing the type of the layout
+	 * @return The pointer to the ffi_type corresponding to the layout symbol
+	 */
+	VMINLINE ffi_type*
+	mapPrimitiveLayoutToFFIType(char layoutSymb)
+	{
+		ffi_type* typeFFI = NULL;
+
+		switch (layoutSymb) {
+		case 'V':	/* VOID */
+			typeFFI = &ffi_type_void;
+			break;
+		case 'C':	/* C_CHAR */
+			typeFFI = &ffi_type_sint8;
+			break;
+		case 'S':	/* C_SHORT */
+			typeFFI = &ffi_type_sint16;
+			break;
+		case 'I':	/* C_INT */
+			typeFFI = &ffi_type_sint32;
+			break;
+		case 'L':	/* either C_LONG or C_LONG_LONG */
+			typeFFI = &ffi_type_sint64;
+			break;
+		case 'F':	/* C_FLOAT */
+			typeFFI = &ffi_type_float;
+			break;
+		case 'D':	/* C_DOUBLE */
+			typeFFI = &ffi_type_double;
+			break;
+		case 'P':	/* C_POINTER */
+			typeFFI = &ffi_type_pointer;
+			break;
+		default:
+			Assert_VM_unreachable();
+		}
+
+		return typeFFI;
+	}
+	
+	/**
+	 * @brief Obtain a primitive FFI type from a layout string object
+	 * @param typeFFI[in] The primitive FFI type to be obtain
+	 * @param layoutStringObject[in] An object containing the layout string describing the target type.
+	 *        e.g. C_INT = b32[abi/kind=INT]
+	 *             C_INT with name c_int = b32(c_int)[abi/kind=INT,layout/name=c_int]
+	 * @return The pointer to the ffi_type corresponding to the layout
+	 */
+	VMINLINE ffi_type*
+	getPrimitiveFFIType(j9object_t layoutStringObject) {
+		PORT_ACCESS_FROM_JAVAVM(_vm);
+
+		ffi_type *primitiveTypeFFI = NULL;
+		char *currentLayout = NULL;
+		char layoutBuffer[J9VM_LAYOUT_STRING_ON_STACK_LIMIT];
+		char *layoutStr = copyStringToUTF8WithMemAlloc(_currentThread, layoutStringObject,
+														J9_STR_NULL_TERMINATE_RESULT,
+														"", 0, layoutBuffer, sizeof(layoutBuffer), NULL);
+		if (NULL == layoutStr) {
+			goto doneGetPrimitiveFFIType;
+		}
+
+		/* The string of layout consists of the endianness ('B' for big-endianness while 'b' for little-endianness),
+		 * the data size and the data type, e.g. b8[abi/kind=CHAR] stands for C_CHAR.
+		 * Thus, we need to skip over '=' to obtain the first character of the type.
+		 */
+		currentLayout = layoutStr;
+		while (*currentLayout != '=') {
+			currentLayout++;
+		}
+		primitiveTypeFFI = mapPrimitiveLayoutToFFIType(*(++currentLayout));
```

Comment:
Please do a +=1 on the previous line, or just use currentLayout[1] since is appears the variable is not used again.

Reply:
Added `currentLayout += 1;` before `mapPrimitiveLayoutToFFIType` as suggested above.